### DISCUSSION TITLE
Should Pinta 2.2 really be Pinta 3.0

### DISCUSSION THREAD

**Discussion by JGCarroll** [PintaProject/Pinta#693] [General]
Pinta 1.7 was a huge update 5 years in the making, but was still seen as a minor patch despite revitalising Pinta for a lot of users.

2.0 was the GTK3 release. If we use Semantic Versioning, it makes sense, since there was some functionality loss (addins) despite the majority being gains.

2.1 was a nice set of improvements over 2.0 but still on GTK3.

For nothing other than vanity and public relations, would it make sense to rebrand the GTK4 port as Pinta 3.0? From my personal observations of the Linux community, they really like GTK4 and see it as huge wins by itself. The Mac/Windows/BSD communities may disagree, but I can't imagine they'd really be bothered about version numbers.

Through nothing other than human psychology, bigger number = better number; and so although rebranding as 3 this late on might mean nothing from a technical POV; it'd be good for drumming up PR on Reddit/etc.

**Comment by cameronwhite** [comment]
I haven't thought much about the version number yet, but those are good arguments :) 
The port to GTK4 / libadwaita (and switching to a different library for the GTK bindings), along with restoring add-in support, has made it at least as large a release as 2.0 with the GTK3 / .NET 6 port

**Reply by cameronwhite** [reply]
As an update to this, I've bumped the version number to 3.0 in 2dbb9d55345f0b4e177b0dba9fe0b373c989e5a4
Since all the known blocker issues are resolved, I'm planning to tag a commit for a beta version which can be used for wider testing (and e.g. for preparing packaging changes for snap / flatpak packages)

**Reply by JGCarroll** [reply]
Fantastic news!

The snap needs no adjustment other than to the source-commit tag, I've just built that tag locally now and it's running fine. I've avoided pushing it to the store because I'm assuming the actual beta release is a bit off for now, let us know when that's the case and it'll be trivial to publish it for wider testing.

**Reply by cameronwhite** [reply]
There is now a `3.0-beta1` tag, so feel free to publish that to the beta channel!

**Reply by JGCarroll** [reply]
Thanks, I've put it on the beta channel now!

**Comment by Lehonti** [comment]
What about increasing the number version when there are major breaking changes in the API? Currently what can be done through extensions is limited by what the API allows, and the API reflects the capabilities of the software, of course (we got a taste of this when I wrote the first non-tileable effect), and in order to expand the things one can do with third-party extensions, the API would basically have to be re-written at some point. Increasing the version number would signal a 'clean break' to both developers and users.

**Reply by JGCarroll** [reply]
I've built a local copy just now of 3.1 and by chance I'd tried to install the demo addins and noticed they didn't work due to a version mismatch (3.1 > 3) - while I don't think the user version string should be tied to API most people don't see, it looks like that's implicitly kind of the case at the moment and there probably is some need to disconnect the Addin API version from the user version so that simple bug fixes to the UI don't instantly invalidate an entire ecosystem and put people in a position they can't upgrade (especially when a lot of package managers are keen to).

Right now it's probably not that important since there's effectively none, but going forward if Addins were expected to thrive, they'd need some guarantees and the initial release of 3.0 being the first release in years to have them seems like an ideal point to make an "Addon API 1.0" officially.

**Reply by cameronwhite** [reply]
Good catch, I've filed bug #1379 for this
With major releases there's probably a decent chance we might need to bump the API version anyways (add-ins have access to a pretty large API surface), but this would definitely be a big problem for minor releases.

**Comment by skironDotNet** [comment]
GTK3 vs GTK4 is a technical aspect for which users don't care in a sense. To me as a user Pinta 3.0 is downgrade, not even a crop function, very minimal tools, it looks like MS Paint, Pinta 3.0 is just different software for a use, you can name it Ginta or Finta (aka fck Pinta 3).
This is mostly personal, but I see a lot of opinions online that this "old" UI makes sense:
<img width="645" height="402" alt="image" src="https://github.com/user-attachments/assets/4ace4095-89dc-4457-8f36-134ff291ffe5" />

**This not so much**  
<img width="1101" height="751" alt="image" src="https://github.com/user-attachments/assets/fa5c2b22-8c48-4022-a82d-a98960683971" />

GTK 3 or 4 who cares are we going back to monochrome TVs? I don't know about other people but my brain encodes shapes + colors to quickly find an option, while looking at white/black shapes is like wtf it  could mean. Skipping colors aspect, 3.0 has much less functionality, it's more like Drawing software

<img width="1071" height="867" alt="image" src="https://github.com/user-attachments/assets/3e318b4f-d94a-4fff-95f5-231240120ec9" />
